lib/services/ios-debug-service.ts

```diff
@@ -0,0 +1,299 @@
+import iOSDevice = require("./../common/mobile/ios/ios-device");
+import net = require("net");
+import path = require("path");
+
+module notification {
+	export function waitForDebug(bundleId: string): string {
+		return bundleId + ":NativeScript.Debug.WaitForDebugger";
+	}
+
+	export function attachRequest(bundleId: string): string {
+		return bundleId + ":NativeScript.Debug.AttachRequest";
+	}
+
+	export function appLaunching(bundleId: string): string {
+		return bundleId + ":NativeScript.Debug.AppLaunching";
+	}
+
+	export function readyForAttach(bundleId: string): string {
+		return bundleId + ":NativeScript.Debug.ReadyForAttach";
+	}
+}
+
+class IOSDebugService implements IDebugService {
+	constructor(
+		private $platformService: IPlatformService,
+		private $iOSEmulatorServices: Mobile.IEmulatorPlatformServices,
+		private $devicesServices: Mobile.IDevicesServices,
+		private $platformsData: IPlatformsData,
+		private $projectData: IProjectData,
+		private $childProcess: IChildProcess,
+		private $logger: ILogger,
+		private $fs: IFileSystem,
+		private $errors: IErrors,
+		private $injector: IInjector,
+        private $npm: INodePackageManager,
+        private $iosXcrunSimctl: IOSXcrunSimctl) {
+	}
+
+	get platform(): string {
+		return "ios";
+	}
+
+	public debug(): IFuture<void> {
+		if ((!options.debugBrk && !options.start) || (options.debugBrk && options.start)) {
+			this.$errors.failWithoutHelp("Expected exactly one of the --debug-brk or --start options.");
+		}
+
+		if (options.emulator) {
+			if (options.debugBrk) {
+				return this.emulatorDebugBrk();
+			} else if (options.start) {
+				return this.emulatorStart();
+			}
+		} else {
+			if (options.debugBrk) {
+				return this.deviceDebugBrk();
+			} else if (options.start) {
+				return this.deviceStart();
+			}
+		}
+
+		this.$errors.failWithoutHelp("Failed to select device or emulator to debug on.");
+	}
+
+	private emulatorDebugBrk(): IFuture<void> {
+		return (() => {
+			var device = this.$iosXcrunSimctl.getRunningEmulator().wait();
+			if (device) {
+				this.$logger.info("Install and run on: " + device.uuid);
+			} else {
+				this.$errors.failWithoutHelp("This command expects a running emulator.");
+			}
+
+			var platformData = this.$platformsData.getPlatformData(this.platform);
+			this.$platformService.buildPlatform(this.platform).wait();
+			var emulatorPackage = this.$platformService.getLatestApplicationPackageForEmulator(platformData).wait();
+
+			device.installApp(emulatorPackage.packageName).wait();
+
+			this.executeOpenDebuggerClient().wait();
+
+			device.launchApp(this.$projectData.projectId, "--nativescript-debug-brk").wait();
+		}).future<void>()();
+	}
+
+	private emulatorStart(): IFuture<void> {
+		return (() => {
+			this.executeOpenDebuggerClient().wait();
+			var device = this.$iosXcrunSimctl.getRunningEmulator().wait();
+			if (!device) {
+				this.$errors.failWithoutHelp("This command expects a running emulator.");
+			}
```

Comment:
Shouldn't  this `this.executeOpenDebuggerClient().wait();` after `if (!device) {`